### ISSUE TITLE
Can't pass signal as create* function parameter

### ISSUE THREAD

**Issue by elsassph** [solidjs-community/eslint-plugin-solid#193]
**Describe the bug**
```typescript
const foo = createSignal(0);

const bar = createSomething({ foo }); // OK
const bar = createSomething(() => foo()); // OK
const bar = createSomething(foo); // <-- solid/reactivity on `foo`
```

**To Reproduce**

**Expected behavior**

**Screenshots**

**Environment (please complete the following information):**

- OS: [e.g. Mac OS 11, Windows 10]
- Node version (`node --version`):
- `eslint-plugin-solid` version (`npm list eslint-plugin-solid`/`yarn why eslint-plugin-solid`): 0.14.5
- `eslint` version (`npm list eslint`/`yarn why eslint`): 8.57.1

**Additional context**

- [ ] I would be willing to contribute a PR to fix this issue

**Comment by ryansolid** [comment]
Confirmed on current main: passing the whole signal tuple (`createSomething(foo)` where `const foo = createSignal(0)`) reports `shouldDestructure`. Passing the tuple wholesale is legitimate API design, so the rule should treat a tuple handed to a function like it treats destructured use. Related: #190 (wrappers around the tuple). On the backlog.

**Comment by ryansolid** [comment]
Verified fixed as of 0.16.1: passing a signal (or memo/projection accessor) as a `create*`/`use*` function argument no longer warns — this landed with the accessor-passing improvements in the 0.16.1 precision pass (see the `createCounter(double)` cases in the reactivity tests). The only message remaining on the original snippet is `shouldDestructure` on `const foo = createSignal(0)`, which is the rule's separate, intentional advice to destructure the tuple so analysis can track getter and setter — not the false positive reported here. Thanks for the report, and sorry it took a while.
